Source/WTF/wtf/Vector.h

```diff
@@ -56,48 +56,34 @@ DECLARE_ALLOCATOR_WITH_HEAP_IDENTIFIER(VectorBuffer);
 
 enum class NulloptBehavior : bool { Ignore, Abort };
 
-template <bool needsDestruction, typename T>
-struct VectorDestructor;
-
-template<typename T>
-struct VectorDestructor<false, T>
-{
-    static void destruct(T*, T*) { }
-};
-
 WTF_ALLOW_UNSAFE_BUFFER_USAGE_BEGIN
 
 template<typename T>
-struct VectorDestructor<true, T>
+struct VectorTypeOperations
 {
-    static void destruct(T* begin, T* end) 
+    static void destruct(T* begin, T* end)
```

Comment:
I don't think we need this at all. I think callers can just used `std::destroy` or `std::ranges::destroy` or if we still want this function for some reason we could use that here instead of writing a loop.